src/core_plugins/kibana/public/discover/controllers/discover.js

```diff
@@ -324,7 +324,13 @@ function discoverController(
         $scope.$listen(queryFilter, 'fetch', $scope.fetch);
 
         $scope.$watch('opts.timefield', function (timefield) {
-          timefilter.enabled = !!timefield;
+          if (!!timefield) {
```

Comment:
I think we could have the auto refresh in discovery enabled even in index patterns without a timefield.
As mentioned in the original ticket, we anyway currently have the "issue", that enabling auto refresh on a timefielded index and then switching to a non time-based index, it still keeps auto refreshing without showing the UI anymore.
I also don't see a reason even if your data isn't time based why it couldn't change and you wouldn't want auto refresh.
@weltenwort your opinion on that?

Reply:
I know this change is trivial but I'd prefer to keep it out of this PR. I'd like this PR to be just about the change in the timefilter API but otherwise keep parity with the state prior to the PR. Could we make follow up PRs that actually take advantage of the new granularity in the API in the places (e.g. Discovery) where it makes sense?

Reply:
I am fine with that.